cpp11/Ice/interceptor/AuthenticatorI.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include <chrono>
+#include <iostream>
+#include <Ice/Ice.h>
+#include <AuthenticatorI.h>
+
+using namespace std;
+
+long long getCurrentTimeMillis()
+{
+    auto duration = chrono::system_clock::now().time_since_epoch();
+    return chrono::duration_cast<chrono::milliseconds>(duration).count();
+}
+
+//
+// Use a value generated from a random device to seed the RNG.
+//
+AuthenticatorI::AuthenticatorI() :
+    _rand(random_device()())
+{
+}
+
+string
+AuthenticatorI::getToken(const Ice::Current&)
+{
+    //
+    // Generate a random 32 character long token.
+    //
+    const int tokenLength = 32;
```

Comment:
I would use a static for chars to avoid allocating the string each time. You could also get rid of the C char array by filling the token string directly:
```
    const int tokenLength = 32;
    static string chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
    uniform_int_distribution<int> dist(0, static_cast<int>(chars.length() - 1));
    string token(tokenLength, '\0');
    for(auto& c : token)
    {
        c = chars.at(dist(_rand));
    }
```

Another possibility would be to have a wider character range: `0-z`:
```
    uniform_int_distribution<> dist('0', 'z');
    string token(tokenLength, '\0');
    for(auto& c : token)
    {
        c = chars.at(dist(_rand));
    }
```